### PR TITLE
completed kata nc-news6 with test

### PR DESCRIPTION
**Pull request by amrutha-chil**

### CODE REVIEW

File: __tests__/nc-news6.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        expect(body.message).toBe("Id Not found");
+      });
+  });
+  test("should return an array of comments for the article id provided", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        console.log(comments);
```
Comment (Slyose) [inline __tests__/nc-news6.test.js]: Don't forget to take out console logs before your PR!

File: __tests__/nc-news6.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        expect(comments[0]).toEqual({
+          comment_id: 145,
+          votes: 10,
+          created_at: "2020-10-03T13:18:00.000Z",
+          author: "jessjelly",
+          body: "Odit aut error. Occaecati et qui. Quam nam aut dolorem.",
+          article_id: 3,
+        });
+      });
+  });
```
Comment (Slyose) [inline __tests__/nc-news6.test.js]: This does work for the first comment - but it'd be a stronger test to go through all the comments and makes sure they have these properties with the right data types. toMatchObject or expect.objectContaining could help us with this!

**Review by Slyose — COMMENTED** [review]
Great job so far. I've left a couple comments to look at to perfect this ticket, feel free to resubmit a PR after you've made any improvements

File: __tests__/nc-news6.test.js (hunk 1 above)
Comment (amrutha-chil) [reply __tests__/nc-news6.test.js]: Done.

File: __tests__/nc-news6.test.js (hunk 2 above)
Comment (amrutha-chil) [reply __tests__/nc-news6.test.js]: Done.

File: __tests__/nc-news6.test.js (hunk 1 above)
Comment (amrutha-chil) [reply __tests__/nc-news6.test.js]: removed the console.log

File: __tests__/nc-news6.test.js (hunk 2 above)
Comment (amrutha-chil) [reply __tests__/nc-news6.test.js]: added toMatchObject

File: __tests__/nc-news6.test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        expect(body.message).toBe("Id Not found");
+      });
+  });
+  test("should return an array of comments for the article id provided", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments[3]).toMatchObject({
```
Comment (mezzDavies) [inline __tests__/nc-news6.test.js]: This is an improvement however we should do this assertion for every object in the array. Can you think of a way to this forEach of the objects 😉 ? If you can you MUST first assert the length of the array is correct before doing so. Please resubmit when you have done this. 👍

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Please see my comment below and resubmit when you have added the code 👍

File: __tests__/nc-news6.test.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  test("should return the most recent comment first", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
```
Comment (Keb84) [inline __tests__/nc-news6.test.js:56]: Another test that would be good here is to check that if the ID is valid but there are no comments it responds with an empty array 💭

**Review by Keb84 — APPROVED** [review]
Great to see you've made the changes suggested. I've suggested one more test for you. Feel free to add that and then merge when ready 👍

**Pull request merged**